src/js/profile/mobile/widget/TimePicker.js

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd
+ *
+ * Licensed under the Flora License, Version 1.1 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://floralicense.org/license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* global define */
+/**
+ * # TimePicker Widget
+ * Shows a control that can be used to set hours and minutes.
+ * It support 12/24 hours format. It contains two inputs which control the values
+ *
+ * ## Default selectors
+ *
+ * Default selector for timepicker is class *ui-time-picker*
+ *
+ * ### HTML Examples
+ *
+ * #### 12 hours format
+ * To add a timepicker widget to the application, use the following code:
+ *
+ *      @example
+ *      <div class="ui-time-picker" data-format="12">
+ *
+ * #### 24 hours format
+ * To add a timepicker widget to the application, use the following code:
+ *
+ *      @example
+ *      <div class="ui-time-picker" data-format="24">
+ *
+ * @class ns.widget.mobile.TimePicker
+ * @since 1.2
+ * @component-selector .ui-time-picker
+ * @extends ns.widget.mobile.TimePicker
+ * @author Hunseop Jeong <hs85.jeong@samsung.com>
+ */
+(function (window, document, ns) {
+	"use strict";
+	//>>excludeStart("tauBuildExclude", pragmas.tauBuildExclude);
+	define(
+		[
+			"../../../core/engine",
+			"../../../core/event",
+			"../../../core/event/gesture",
+			"./BaseWidgetMobile",
+			"./Spin"
+		],
+		function () {
+			//>>excludeEnd("tauBuildExclude");
+			var BaseWidget = ns.widget.mobile.BaseWidgetMobile,
+				engine = ns.engine,
+				Spin = ns.widget.mobile.Spin,
+				WIDGET_CLASS = "ui-time-picker",
+				classes = {
+					CONTAINER: WIDGET_CLASS + "-container",
+					HOUR_CONTAINER: WIDGET_CLASS + "-container-hour",
+					MINUTE_CONTAINER: WIDGET_CLASS + "-container-minute",
+					FORMAT_CONTAINER: WIDGET_CLASS + "-container-format"
+				},
+				WIDGET_SELECTOR = "." + WIDGET_CLASS,
+				TimePicker = function () {
+					this.options = {
+						format: "12"
+					};
+
+					this._ui = {};
+					this._spins = {
+						hour: null,
+						minute: null,
+						format: null
+					};
+				},
+				prototype = new BaseWidget();
+
+			TimePicker.classes = classes;
+
+			/**
+				* Init method
+				* @method _init
+				* @member ns.widget.mobile.TimePicker
+				* @protected
+				*/
+			prototype._init = function () {
+				var self = this,
+					options = self.options,
+					initDate = new Date();
+
+				options.format = (options.format !== undefined) ? options.format : "12";
+
+				self._setValue(initDate);
```

Comment:
question.
when init, does setValue set to the current time?
Is there any reason to differentiate between setValue and setDateValue?

Reply:
Yes. It set the value with current time.
I will extend the setValue to set the time with Date Object, Object or value.
Currently, I impelmented it to set with Date Object only.